server.go

```diff
@@ -203,6 +212,9 @@ func (fn ErrorHandlerFunc) HandleError(ctx context.Context, task *Task, err erro
 	fn(ctx, task, err)
 }
 
+// BaseCtxFn provides the root context from where the execution contexts of tasks are derived
```

Comment:
nit: I don't think we need to declare this type. Just spell this out for better readability.